persist/sqldb/workflow_archive.go

```diff
@@ -257,14 +257,43 @@ func namePrefixClause(namePrefix string) db.Cond {
 	}
 }
 
-func (r *workflowArchive) GetWorkflow(uid string) (*wfv1.Workflow, error) {
+func (r *workflowArchive) GetWorkflow(uid string, namespace string, name string) (*wfv1.Workflow, error) {
+	var err error
 	archivedWf := &archivedWorkflowRecord{}
-	err := r.session.
-		Select("workflow").
-		From(archiveTableName).
-		Where(r.clusterManagedNamespaceAndInstanceID()).
-		And(db.Cond{"uid": uid}).
-		One(archivedWf)
+	if uid != "" {
+		err = r.session.
+			Select("workflow").
+			From(archiveTableName).
+			Where(r.clusterManagedNamespaceAndInstanceID()).
+			And(db.Cond{"uid": uid}).
+			One(archivedWf)
+	} else {
+		if name != "" && namespace != "" {
+			total := &archivedWorkflowCount{}
+			err = r.session.
+				Select(db.Raw("count(*) as total")).
+				From(archiveTableName).
+				Where(r.clusterManagedNamespaceAndInstanceID()).
+				And(namespaceEqual(namespace)).
+				And(nameEqual(name)).
+				One(total)
+			if err != nil {
+				return nil, err
+			}
+			if int64(total.Total) > 1 {
+				return nil, fmt.Errorf("there are more than 1 workflows with the same name and namespace")
+			}
+			err = r.session.
+				Select("workflow").
+				From(archiveTableName).
+				Where(r.clusterManagedNamespaceAndInstanceID()).
+				And(namespaceEqual(namespace)).
+				And(nameEqual(name)).
+				One(archivedWf)
+		} else {
+			return nil, fmt.Errorf("both name and namespace are required if uid is not specified")
```

Comment:
Is there a way to return an error that would cause us to return 400 Bad Request as an http response?

Reply:
Yes, see my latest commit.

Reply:
The mapping can be found in https://github.com/argoproj/argo-workflows/blob/master/server/utils/errors.go